docs/faq/ImmutableData.md

```diff
@@ -0,0 +1,245 @@
+- [Is Immutable.JS worth the effort?](#is-immutable-js-worth-effort)
+- [What are the Recommended Best Practices for using Immutable.JS with Redux?](#immutable-js-best-practices)
+
+<a id="benefits-of-immutability"></a>
+## What are the benefits of immutability?
+
+Immutability can bring increased performance to your app, and leads to simpler programming and debugging, as data that never changes is easier to reason about than data that is free to be changed arbitrarily throughout your app.
+
+In particular, immutability in the context of a Web app enables sophisticated change detection techniques to be implemented simply and cheaply, ensuring the computationally expensive process of updating the DOM occurs only when it absolutely has to (a cornerstone of React’s performance improvements over other libraries).
+<a id="why-is-immutability-required"></a>
+## Why is immutability required in Redux?
+
+In the context of Redux, if your reducers mutate their arguments and directly modify the state tree, you will cause React Redux’s `connect` function to break, as it will be unable to detect changes made to objects and arrays. You MUST therefore use immutable operations in your reducers when using Redux.
+<a id="do-i-have-to-use-immutable-js"></a>
+## Do I have to use Immutable.JS?
+
+You do not need to use Immutable.JS with Redux. ES6, if written correctly, is perfectly capable of providing immutability without having to use an immutable-focused library. However, guaranteeing immutability with ES6 is difficult, and it can be easy to mutate an object accidentally, causing bugs in your app that are extremely difficult to locate. For this reason, using a library such as Immutable.JS can significantly improve the reliability of your app, and make your app’s development much easier.
+<a id="issues-with-es6-for-immutable-ops"></a>
+## What are the issues with using ES6 for immutable operations?
+ES6 was never designed to provide guaranteed immutable operations. Accordingly, there are several issues you need to be aware of if you choose to use ES6 for your Immutable operations in your Redux app.
+#### Accidental Object Mutation
+With ES6, you can accidentally mutate an object (such as the Redux state tree) quite easily without realising it. For example, updating deeply nested properties, creating a new *reference* to an object instead of a new object, or performing a shallow copy rather than a deep copy, can all lead to inadvertent object mutations, and can trip up even the most experienced JavaScript coder. 
+
+To avoid these issues, ensure you follow the recommended [Immutable Update Patterns for ES6](http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html).
+
+#### Verbose Code
+Updating complex nested state trees can lead to verbose code that is tedious to write and difficult to debug.
+
+#### Poor Performance
+Operating on ES6 objects and arrays in an immutable way can be slow, particularly as your state tree grows larger.
```

Comment:
Might want to clarify _why_ this is slow - I believe it has to do with copying properties from object to object, although it would be good to point to some benchmarks on that.